faunadb-scala/src/test/scala/faunadb/ClientSpec.scala

```diff
@@ -2229,6 +2231,50 @@ class ClientSpec
     }
   }
 
+  it should "parallel testing should be executed without exceptions" in {
+    val key = rootClient.query(CreateKey(Obj("database" -> Database(testDbName), "role" -> "admin"))).futureValue
+    val clientPool = List.tabulate(10)(n=> FaunaClient(endpoint = config("root_url"), secret = key(SecretField).get))
+
+    val random = scala.util.Random
+    val COLLECTION_NAME = "ParallelTestCollection"
+    //create sample collection with 10 documents
+    client.query(CreateCollection(Obj("name" -> COLLECTION_NAME))).futureValue
+    for (i <- 1 to 3) {
+      client.query(
+          Create(Collection(COLLECTION_NAME),
+            Obj("data" -> Obj("testField" -> "testValue")))).futureValue
+    }
+
+    val numJobs = 500
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
```

Comment:
I don't have much experience in Scala and can be mistaken, but does this line does anything?

Reply:
this is thread pool size. I think 10 threads are ok for this task